components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/model/ActionType.java

```diff
@@ -18,11 +18,24 @@
 
 package org.wso2.carbon.identity.action.execution.model;
 
+import org.wso2.carbon.identity.action.management.model.Action;
+
 /**
  * This class models the Action Type.
  * Action Type is the type of the action that is executed by the Action Executor Service.
  */
 public enum ActionType {
-    PRE_ISSUE_ACCESS_TOKEN,
-    AUTHENTICATION
+    PRE_ISSUE_ACCESS_TOKEN("PRE_ISSUE_ACCESS_TOKEN"),
+    AUTHENTICATION("AUTHENTICATION");
+
+    private final String value;
+
+    ActionType(String value) {
+        this.value = value;
+    }
+
+    public String getDisplayName() {
+
+        return Action.ActionTypes.valueOf(this.value).getDisplayName();
```

Comment:
we don't need this string instance variable right ? why not just use this.name()

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/6056/commits/762868948807a4e14cd411a8c9b1d1ba9243de89